validator/sawtooth_validator/journal/consensus/dev_mode/dev_mode_consensus.py

```diff
@@ -21,51 +24,87 @@
 
         self._block_cache = block_cache
         self._state_view = state_view
-        self.batch_publisher = batch_publisher
-        self._num_batches = 1
-        self._count = 0
+
+        self._start_time = 0
+        self._wait_time = 0
+        config_view = ConfigView(self._state_view)
+        self._min_wait_time = config_view.get_setting(
+            "sawtooth.consensus.min_wait_time", 0, int)
+        self._max_wait_time = config_view.get_setting(
+            "sawtooth.consensus.max_wait_time", 0, int)
+        self._valid_block_publishers = config_view.get_setting(
+            "sawtooth.consensus.valid_block_publishers", None, list)
 
     def initialize_block(self, block_header):
-        """Initialize the candidate block_header.
+        """Do initialization necessary for the consensus to claim a block,
+        this may include initiating voting activates, starting proof of work
+        hash generation, or create a PoET wait timer.
+
         Args:
-            block_header (block_header): the block_header to initialize.
+            journal (Journal): the current journal object
+            block (TransactionBlock): the block to initialize.
         Returns:
-            Boolean: True if the candidate block_header should be built.
+            none
         """
+        block_header.consensus = b"Devmode"
+        self._start_time = time.time()
+        self._wait_time = random.uniform(
+            self._min_wait_time, self._max_wait_time)
         return True
 
-    def check_publish_block(self, block_header):
-        """Check if a candidate block_header is ready to be claimed.
+    def check_publish_block(self, block):
+        """Check if a candidate block is ready to be claimed.
+
         Args:
-            block_header: the block_header to be checked if it
-            should be claimed.
+            journal (Journal): the current journal object
+            block: the block to be checked if it should be claimed
         Returns:
             Boolean: True if the candidate block_header should be claimed.
         """
-        if self._count >= self._num_batches:
-            self._count = 0
+        if self._valid_block_publishers\
+                and block.block_header.signer_pubkey \
+                not in self._valid_block_publishers:
+            return False
+        elif self._min_wait_time == 0:
             return True
-        self._count += 1
-        return False
-
-    def finalize_block(self, block_header):
-        """Finalize a block_header to be claimed.
+        elif self._min_wait_time > 0 and self._max_wait_time <= 0:
+            if self._start_time + self._min_wait_time <= time.time():
+                return True
+        elif self._min_wait_time > 0 \
+                and self._max_wait_time > self._min_wait_time:
+            if self._start_time + self._wait_time <= time.time():
+                return True
+        else:
+            return False
+
+    def finalize_block(self, block):
```

Comment:
Same issue with the arguments.  Take a look at `validator/sawtooth_validator/journal/consensus.py` for the interface contract.